src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -54,21 +63,37 @@
     List<Object> unmergedFields = new ArrayList<>();
     for (String fieldSpec : fieldSpecs) {
       Object topLevel = InitValueConfig.create();
-      if (initValueConfigMap.containsKey(fieldSpec)) {
+      String[] equalsParts = fieldSpec.split("[=]");
+      if (equalsParts.length > 2) {
+        throw new IllegalArgumentException("Inconsistent: found multiple '=' characters");
+      }
+      if (equalsParts.length == 2) {
+        topLevel = InitValueConfig.createWithValue(equalsParts[1]);
+      } else if (initValueConfigMap.containsKey(fieldSpec)) {
         topLevel = initValueConfigMap.get(fieldSpec);
       }
-      String[] stringParts = fieldSpec.split("[.]");
+      String[] stringParts = equalsParts[0].split("[.]");
       for (int i = stringParts.length - 1; i >= 0; i--) {
-        Matcher listMatcher = fieldListPattern.matcher(stringParts[i]);
-        if (listMatcher.matches()) {
-          String name = listMatcher.group(1);
-          String index = listMatcher.group(2);
-          ListElementSpec listStructure = new ListElementSpec(index, topLevel);
-          topLevel = new FieldSpec(name, listStructure);
-          // TODO also support maps - use {key} notation
-        } else {
-          String name = stringParts[i];
-          topLevel = new FieldSpec(name, topLevel);
+        String toMatch = stringParts[i];
+        while (true) {
```

Comment:
The purpose is to recursively strip [] and {} from the end of the string. I will refactor.